torch/csrc/jit/fusion_compiler.h

```diff
@@ -60,6 +60,15 @@ struct AnnotatedGraph {
   std::vector<TensorDesc> output_desc;
 };
 
+struct ChunkDesc {
```

Comment:
Now that I think about this more, I believe ChunkDesc and ConcatDesc are actually exactly the same thing. Each describes how to split up a bigger tensor into smaller components that become inputs into the kernel. The only difference is that for chunks it describes an input, and for cat, an output. However from the kernels perspective they are all just pointers to memory anyway. It should cleaner to actually have the code reflect this similarity, and because they are the same concept, they should also be able to share a lot of code. For instance I was suggesting adding a precomputed TensorDesc to Chunk -- well concat already has that and the handling for it.

Reply:
Good point! I'll look into using ConcatDesc and sharing code between chunk/cat